qos/evm/service_state.go

```diff
@@ -0,0 +1,278 @@
+package evm
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+
+	"github.com/buildwithgrove/path/gateway"
+	"github.com/buildwithgrove/path/protocol"
+	"github.com/buildwithgrove/path/qos/jsonrpc"
+)
+
+// The serviceState struct maintains the expected current state of the EVM blockchain
+// based on the endpoints' responses to different requests.
+//
+// It has three main responsibilities:
+//  1. Generate QoS endpoint checks for the hydrator
+//  2. Select a valid endpoint for a service request
+//  3. Update the service state from endpoint observations
```

Comment:
3. Update the stored endpoints from observations.
4. Update the stored service state from observations.